test/parallel/test-tls-inception.js

```diff
@@ -33,7 +32,7 @@ a = tls.createServer(options, function(socket) {
   dest.pipe(socket);
   socket.pipe(dest);
 
-  dest.on('close', function() {
+  dest.on('end', function() {
```

Comment:
I've tried to explain it the best I can (from my understanding) in the commit log: https://github.com/santigimeno/node/commit/432a86eb88f69e8d94253087bfd707cfce73828c
```
When sending a very large buffer (400000 bytes) the test fails due to
the client socket from the `a` server erroring with `ECONNRESET`.
There's a race condition between the closing of this socket and the `ssl`
socket closing on the other side of the connection. To improve things,
destroy the socket as soon as possible: in the `end` event of the `dest`
socket.
```
Running the test with `dtruss` I could see the `ssl` closing and the `socket` sending data afterwards. This data was from the `SSL_shutdown` called when the `dest` socket was closed (I guess because both sockets were piped)
What I don't really understand is how `ssl` receives an `EOF`, thus emitting the `end` event, before the other end has sent the shutdown. (I'm not sure if it makes sense).